events/events.go

```diff
@@ -8,48 +8,74 @@
 // by the Apache License, Version 2.0, included in the file
 // licenses/APL.txt.
 
-/*
-Package events provides the internal event system.
-*/
 package events
 
-type Subscription[T any] chan T
+import (
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p/core/event"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
 
-// Channel represents a subscribable type that will expose inputted items to subscribers.
-type Channel[T any] interface {
-	// Subscribe subscribes to the Channel, returning a channel by which events can
-	// be read from, or an error should one occur (e.g. if this object is closed).
-	//
-	// This function is non-blocking unless the subscription-buffer is full.
-	Subscribe() (Subscription[T], error)
+const (
+	// WildCardEventName is the alias used to subscribe to all events.
+	WildCardEventName = "*"
+	// MergeCompleteEventName is the name of the database merge complete event.
+	MergeCompleteEventName = "db:merge-complete"
+	// UpdateEventName is the name of the database update event.
+	UpdateEventName = "db:update"
+	// ResultsEventName is the name of the database results event.
+	ResultsEventName = "db:results"
+	// MergeRequestEventName is the name of the net merge request event.
+	MergeRequestEventName = "net:merge"
+	// PubSubEventName is the name of the network pubsub event.
+	PubSubEventName = "net:pubsub"
+	// ConnectEventName is the name of the network connect event.
+	ConnectEventName = "net:connect"
+)
 
-	// Unsubscribe unsubscribes from the Channel, closing the provided channel.
-	//
-	// Will do nothing if this object is already closed.
-	Unsubscribe(Subscription[T])
+// ConnectEvent is an event that is published when
+// a peer connection has changed status.
+type ConnectEvent = event.EvtPeerConnectednessChanged
```

Comment:
suggestion: Since we are in the `events` package, the `Event` suffix isn't necessary. We could change `events.ConnectEvent`, `events.PubSubEvent`, `events.UpdateEvent`, `events.MergeEvent` to `events.Connect`, `events.PubSub`, `events.Update`, `events.Merge`

Reply:
that would be neat. But for this, following best go practices, would be better to make package name singular `event`.

Reply:
@nasdf you changed to `event` but didn't do my suggestion. Did you just forget or did you decide you didn't like it? :)

Reply:
@fredcarle I'm considering moving the event definitions to their respective packages.

Reply:
Also a good option 👍

Reply:
The downside of defining the events in their respective packages is that if might result in unwanted dependencies or even circular dependencies depending in which package it is defined. `MergeEvent` being defined in `db` or `net` for example.

Reply:
Good point. I've kept them in the package but renamed as you suggested.

Reply:
this is being discussed in another thread https://github.com/sourcenetwork/defradb/pull/2700#discussion_r1633762423 :)